doc/tutorials/acrn-secure-boot-with-grub.rst

```diff
@@ -0,0 +1,179 @@
+.. _how-to-enable-acrn-secure-boot-with-grub:
+
+Enable ACRN Secure Boot with GRUB
+#################################
+
+This document shows how to enable ACRN secure boot with GRUB including:
+
+  -  ACRN Secure Boot Sequence
+  -  Generate GPG Key
+  -  Setup Standalone GRUB EFI Binary
+  -  UEFI Secure Boot
+
+  Validation Environment:
+	Hardware Platform: WHL-IPC-I7, Supported hardware described in :ref:`hardware`.
+	ACRN Scenario: Industry
+	Service VM: Ubuntu 18.04
+	GRUB: 2.02
+
+..note::
+	Note that GRUB may stop booting in case of problems, make sure you know how to recover a bootloader on your platform.
+
+ACRN Secure Boot Sequence
+*************************
+
+  ACRN can be booted by Multiboot compatible bootloader, following diagram illustrates the boot sequence of ACRN with GRUB:
+
+     .. image:: images/acrn_secureboot_flow.png
+        :align: center
+        :width: 600px
+
+
+  Refer to :ref:`using_grub` for details to enable GRUB on ACRN.
+
+  From point view of secureboot:
+	- UEFI firmware verifies Shim/GRUB
+	- GRUB verifies ACRN, Service VM kernel and pre-launched User VM kernel
+	- Service VM OS kernel verifies the Device Model (`acrn-dm`) and User VM OVMF bootloader (with the help of `acrn-dm`)
+	- User VM virtual bootloader (e.g. OVMF) starts the guest side verified boot process
+
+   This document provides guide to enable verification function of GRUB to verify ACRN binaries (such `acrn.bin`, Service VM kernel (`bzImage`) and pre-launched User VM kernel image if it is present).
+
+
+Generate GPG Key
+****************
+
+  GRUB supports loading of GPG signed files only if digital signatures enabled.
+  Example to generate GPG signing key::
+  $ mkdir --mode 0700 keys
+  $ gpg --homedir keys  --gen-key
+  $ gpg --homedir keys --export > boot.key
+
+  ``gpg --gen-key`` generates key pair, private key will be used to sign GRUB configuration files and ACRN binaries, public key will be embedded in GRUB and is used to verify GRUB configuration files or binaries GRUB trys to load.
+
+
+Setup Standalone GRUB EFI Binary
+********************************
+
+Prepare Initial GRUB Configuration grub.init.cfg
+================================================
+  Create file ``grub.init.cfg`` to store the following minimal grub configuration, ``set check_signatures=enforce`` is to enable digital signatures in GRUB::
```

Comment:
@dbkinder can provide more details but ReST is sensitive to indentation, I'm wondering what it does when it sees that sentence starting with three spaces.